backends/dnf/pk-backend-dnf.c

```diff
@@ -1815,12 +1671,22 @@ pk_backend_refresh_cache_thread (PkBackendJob *job,
 		}
 
 		/* check and download */
-		state_loop = dnf_state_get_child (state_local);
-		ret = pk_backend_refresh_repo (job, repo, state_loop, &error);
-		if (!ret) {
-			pk_backend_job_error_code (job, error->code, "%s", error->message);
-			return;
-		}
+		refresh[0] = LIBEXECDIR "/packagekit-refresh-repo";
```

Comment:
If this is DNF specific, we probably should name the binary accordingly. `packagekit-dnf-refresh-repos` is probably a better name here.

Reply:
Done.